### ISSUE TITLE
VAE allows only CGaussian

### ISSUE THREAD

**Issue by smidl** [nmheim/GenerativeModels.jl#40]
Can we allow VAE for AbrstactPDF and AbstractCPDF? As it is now, I can not use SharedVarCGaussian in VAE.

**Comment by nmheim** [comment]
Would it still be a VAE if it was an AbstractCPDF? We also have:

```julia
abstract type AbstractCPDF{T<:Real} end
abstract type AbstractCGaussian{T} <: AbstractCPDF{T} end
struct CGaussian{T,V<:AbstractVar} <: AbstractCGaussian{T} end
# same for SharedVarCGaussian
```

so we could also make it a `AbstractCGaussian`. I can change it to either one.

**Comment by nmheim** [comment]
closed with #36
